tests/test_transform.py

```diff
@@ -668,13 +628,114 @@ def checkGetJacobian(self, fromName, toName):
+            the prefix of the ending endpoint for the final, concatenated
+            Transform
+        """
+        transform1Class = getattr(afwGeom,
+                                  "Transform{}To{}".format(fromName, midName))
+        transform2Class = getattr(afwGeom,
+                                  "Transform{}To{}".format(midName, toName))
+        baseMsg = "{}.of({})".format(transform2Class.__name__,
+                                     transform1Class.__name__)
+        for nIn in self.goodNaxes[fromName]:
+            for nMid in self.goodNaxes[midName]:
+                for nOut in self.goodNaxes[toName]:
+                    msg = "{}, nIn={}, nMid={}, nOut={}".format(
+                        baseMsg, nIn, nMid, nOut)
+                    polyMap = makeTwoWayPolyMap(nIn, nMid)
+                    transform1 = transform1Class(polyMap)
+                    polyMap = makeTwoWayPolyMap(nMid, nOut)
+                    transform2 = transform2Class(polyMap)
+                    transform = transform2.of(transform1)
+
+                    fromEndpoint = transform1.getFromEndpoint()
+                    toEndpoint = transform2.getToEndpoint()
+
+                    inPoint = fromEndpoint.pointFromData(makeRawPointData(nIn))
+                    outPointMerged = transform.tranForward(inPoint)
+                    outPointSeparate = transform2.tranForward(
+                        transform1.tranForward(inPoint))
+                    assert_allclose(toEndpoint.dataFromPoint(outPointMerged),
+                                    toEndpoint.dataFromPoint(outPointSeparate),
+                                    err_msg=msg)
+
+                    outPoint = toEndpoint.pointFromData(makeRawPointData(nOut))
+                    inPointMerged = transform.tranInverse(outPoint)
+                    inPointSeparate = transform1.tranInverse(
+                        transform2.tranInverse(outPoint))
+                    assert_allclose(
+                        fromEndpoint.dataFromPoint(inPointMerged),
+                        fromEndpoint.dataFromPoint(inPointSeparate),
+                        err_msg=msg)
+
+        # Mismatched number of axes should fail
+        if midName == "Generic":
+            nIn = self.goodNaxes[fromName][0]
+            nOut = self.goodNaxes[toName][0]
+            polyMap = makeTwoWayPolyMap(nIn, 3)
+            transform1 = transform1Class(polyMap)
+            polyMap = makeTwoWayPolyMap(2, nOut)
+            transform2 = transform2Class(polyMap)
+            with self.assertRaises(InvalidParameterError):
+                transform = transform2.of(transform1)
+
+        # Mismatched types of endpoints should fail
+        if fromName != midName:
+            joinNaxes = set(self.goodNaxes[fromName]).intersection(
+                self.goodNaxes[midName])
+            for nIn in self.goodNaxes[fromName]:
+                for nMid in joinNaxes:
+                    for nOut in self.goodNaxes[midName]:
```

Comment:
I'm keeping this subtest simple by making both Transforms the same type, but I agree that it's hard to spot. Any suggestions for how to make it more obvious?